### PR TITLE
Issue:3174324 pdr plugin support

### PR DESCRIPTION
**Pull request by drorlevy**

### CODE REVIEW

**Comment by drorlevy** [comment]
bot:retest

File: plugins/pdr_deterministic_plugin/ufm_sim_web_service/ufm_communication_mgr.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        #self.suffix = None
+        self._host = "{0}:{1}".format(host, self.internal_port)
+    
+    def get_request(self, uri, headers=None):
+        request = self.ufm_protocol + '://' + self._host + uri
+        if not headers:
+            headers = self.headers
+        response = requests.get(request, verify=False, headers=headers)
+        logging.info("UFM API Request Status: {}, URL: {}".format(response.status_code, request))
+        if response.status_code == 200:
```
Comment (kedeme) [inline plugins/pdr_deterministic_plugin/ufm_sim_web_service/ufm_communication_mgr.py:35]: use HTTP_OK constant

**Review by kedeme — CHANGES_REQUESTED** [review]

File: plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py [hunk 2]
```diff
@@ -199,14 +206,53 @@ def read_next_set_of_high_ber_or_pdr_ports(self):
...
+    def get_ports_metadata(self):
+        meta_data = self.ufm_client.get_ports_metadata()
+        if meta_data and len(meta_data) > 0:
+            for port in meta_data:
+                port_name = port.get(Constants.PORT_NAME)
+                if not self.ports_data.get(port_name):
+                    self.ports_data[port_name] = {}
+                self.ports_data[port_name][Constants.ACTIVE_SPEED] = port.get(Constants.ACTIVE_SPEED)
+
+    def get_port_metadata(self, port_name):
```
[the remark this answers is not in the document]
Comment (drorlevy) [reply plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py:325]: but it's also the width and hw_technology

File: plugins/pdr_deterministic_plugin/build/config/fec_mode_lookup.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
```
[the remark this answers is not in the document]
Comment (drorlevy) [reply plugins/pdr_deterministic_plugin/build/config/fec_mode_lookup.json:1]: there isn't such a location. maybe in the arch documents of the firmware. this code is taken from the telemetry team

File: plugins/pdr_deterministic_plugin/build/config/supervisord.conf [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+user=root
+nodaemon = true
+environment = PLACEHOLDER=true
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stderr
+stderr_logfile_maxbytes=0
+
+[program:ufm_web_service]
+command=python3 /opt/ufm/ufm_plugin_pdr_deterministic/ufm_sim_web_service/isolation_algo.py
```
[the remark this answers is not in the document]
Comment (drorlevy) [reply plugins/pdr_deterministic_plugin/build/config/supervisord.conf:19]: it is to keep with the convention of other plugins

File: plugins/pdr_deterministic_plugin/build/docker_build.sh [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+if [ "$EUID" -ne 0 ]
+  then echo "Please run the script as root"
+  exit
+fi
+
+SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"
+
+IMAGE_VERSION=$1
+IMAGE_NAME=ufm-plugin-pdr_deterministic
```
[the remark this answers is not in the document]
Comment (drorlevy) [reply plugins/pdr_deterministic_plugin/build/docker_build.sh:13]: packet drop rate? i kept it as the name of the feature but up to you

**Review by kedeme — APPROVED** [review]

**Pull request merged**